docs/cudf/source/developer_guide/benchmarking.md

```diff
@@ -0,0 +1,288 @@
+Therefore, we have separate benchmarks for `DataFrame`, in addition to the general benchmarks for all `Frame` and `Index` classes.
+
+```{note}
+`pytest` does not support having two benchmark files with the same name, even if they are in separate directories.
+Therefore, benchmarks of internal methods of _public_ classes go in files suffixed with `_internal`.
+Benchmarks of `DataFrame._apply_boolean_mask`, for instance, belong in `internal/bench_dataframe_internal.py`.
+```
+
+Free functions have more flexibility.
+Broadly speaking, they should be grouped into benchmark files containing similar functionality.
+For example, I/O benchmarks can all live in `bench_io.py`.
+For now those groupings are left to the discretion of developers.
+
+## Running benchmarks
+
+The main distinction between tests and benchmarks is configured prefix.
+Bench files and functions are prefixed with `bench_` rather than `test_`.
+`pytest` is automatically configured to find benchmarks this way, though,
+so running benchmarks is as simple as installing `pytest-benchmark` and then running `pytest`.
+
+When benchmarks are run, the default behavior is to output the results in a table to the terminal.
+A common requirement is to then compare the performance of benchmarks before and after a change.
+We can generate these comparisons by saving the output using the `--benchmark-autosave` option to pytest.
+When using this option, after the benchmarks are run the output will contain a line:
+```
+Saved benchmark data in: /path/to/XXXX_*.json
+```
+
+The `XXXX` is a four-digit number identifying the benchmark.
+If preferred, a user may also use the `--benchmark-save=NAME` option,
+which allows more control over the resulting filename.
+Given two benchmark runs `XXXX` and `YYYY`, benchmarks can then be compared using
+```
+pytest-benchmark compare XXXX YYYY
+```
+Note that the comparison uses the `pytest-benchmark` command rather than the `pytest` command.
+`pytest-benchmark` has a number of additional options that can be used to customize the output.
+The next line contains one useful example, but developers should experiment to find a useful output
+```
+pytest-benchmark compare XXXX YYYY --sort="name" --columns=Mean --name=short --group-by=param
+```
+
+## Benchmark contents
+
+### Benchmark configuration
+
+Our benchmarks aim to support [comparing to pandas](pandascompare) or [running tests](testing) out of the box.
+In order for the former to work, _all tests must import `cudf` and `cupy` from the `config` module_.
+In other words:
+```python
+from ..common.config import cudf  # This is good
+import cudf  # This is bad
+```
+Testing is usually transparently supported except when users define custom fixtures or cases.
+In those instances, as a general rule developers should avoid hardcoding data sizes in benchmarks.
+Instead, data sizes should be dependent on variables stored in `config.py`.
+This requirement is discussed in more depth below.
```

Comment:
```suggestion
Benchmarks should support both [comparing to pandas](pandascompare) and [being run as tests](testings) out of the box. To satisfy these requirements, one must take a small amount of care when writing a benchmark.

1. Import `cudf` and `cupy` from the config module (this enables swapping out for `pandas` and `numpy` respectively):
    ```python
        from ..common.config import cudf, cupy # Do this
        import cudf, cupy # Not this
    ```
2. Avoid hard-coding benchmark dataset sizes, and instead use the sizes advertised by `config.py` (this enables running the benchmarks in "test" mode on small datasets, which can be much faster)
```

Reasoning: provide the minimal amount of information for what to do (and a short rationale) up front for those people (like me) who are very bad at reading to the end of the documentation page.

Reply:
Good point. I've adopted your suggestion with some very minor modifications.